lib/safe-pg-migrations/plugins/statement_insurer.rb

```diff
@@ -98,6 +98,29 @@ def change_column_null(table_name, column_name, null, default = nil)
       remove_check_constraint table_name, "#{column_name} IS NOT NULL"
     end
 
+    def remove_column(table_name, column_name, *)
+      foreign_key = foreign_key_for(table_name, column: column_name)
+
+      with_setting(:statement_timeout, SafePgMigrations.config.pg_statement_timeout) do
+        remove_foreign_key(table_name, name: foreign_key.name) if foreign_key
+        super
+      end
+    end
+
+    ruby2_keywords def drop_table(table_name, *args)
+      foreign_keys(table_name).each do |foreign_key|
+        with_setting(:statement_timeout, SafePgMigrations.config.pg_statement_timeout) do
+          remove_foreign_key(table_name, name: foreign_key.name)
```

Comment:
Does it include multiple retry?

Reply:
Yes : https://github.com/doctolib/safe-pg-migrations/blob/d76d22ea88cf096f02083f77e8b5bad069e6f033/lib/safe-pg-migrations/plugins/statement_retrier.rb#L5-L14

It is hire in the inheritance chain, so each call to remove_foreign_key and drop_table will be retried `SafePgMigrations.config.max_tries` times